### PR TITLE
Fixed user.present / user.absent in windows

### PR DESCRIPTION
**Pull request by twangboy**
Fixes: #28928
Updated `states/user.py` to work with the backported win_useradd.py
This is a regression caused by: #28627

### CODE REVIEW

File: salt/states/user.py [hunk 1]
```diff
@@ -490,22 +567,41 @@ def present(name,
...
-                                fullname=fullname,
-                                roomnumber=roomnumber,
-                                workphone=workphone,
-                                homephone=homephone,
-                                loginclass=loginclass,
-                                createhome=createhome):
+
+        # Setup params specific to Linux and Windows to be passed to the
+        # add.user function
+        if not salt.utils.is_windows():
```
Comment (thedrow) [inline salt/states/user.py]: Can you extract this section into a helper function? This will help unit testing a lot.

**Comment by thedrow** [comment]
Can we adjust the existing tests to check this? I am aware that the CI doesn't run a build step on Windows (see https://github.com/saltstack/salt-windows-dev/issues/12 and #20239) but when it will we'd like to verify that this still works correctly.

**Comment by cachedout** [comment]
@twangboy Are you gonig to respond to the comments on this PR from @thedrow ? Please let me know if this is ready to be merged.

**Comment by thedrow** [comment]
Let's label this so we won't forget.

**Pull request merged**